cmd/state-svc/internal/deprecation/deprecation.go

```diff
@@ -0,0 +1,200 @@
+package deprecation
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	constvers "github.com/ActiveState/cli/internal/constants/version"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/graph"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/retryhttp"
+	"github.com/hashicorp/go-version"
+	"github.com/patrickmn/go-cache"
+)
+
+const (
+	cacheKey = "info"
+)
+
+type ErrTimeout struct {
+	error
+}
+
+// info details deprecation information for a given version
+type info struct {
+	Version     string `json:"version"`
+	versionInfo *version.Version
+	Date        time.Time `json:"date"`
+	DateReached bool
+	Reason      string `json:"reason"`
+}
+
+// Checker is the struct that we use to do checks with
+type Checker struct {
+	config configurable
+	cache  *cache.Cache
+	done   chan struct{}
+}
+
+// configurable defines the configuration function used by the functions in this package
+type configurable interface {
+	ConfigPath() string
+	GetTime(key string) time.Time
+	Set(key string, value interface{}) error
+	Close() error
+}
+
+// NewChecker returns a new instance of the Checker struct
+func NewChecker(configuration configurable) *Checker {
+	checker := &Checker{
+		configuration,
+		cache.New(15*time.Minute, 5*time.Minute),
+		make(chan struct{}),
+	}
+
+	go checker.pollDeprecationInfo()
+
+	return checker
+}
+
+// Check will Check if the current version of the tool is deprecated and returns deprecation info if it is.
+// This uses a fairly short timeout to Check against our deprecation url, so this should not be considered conclusive.
+func (checker *Checker) Check() (*graph.DeprecationInfo, error) {
+	data, exists := checker.cache.Get(cacheKey)
+	if !exists {
+		return nil, nil
```

Comment:
Since in THEORY this should never happen we should error log this.

Reply:
When the change that is suggested [here](https://github.com/ActiveState/cli/pull/1796#discussion_r854577055) this will happen often so I've just made it a debug log.